support/rbac.go

```diff
@@ -202,3 +202,37 @@ func CreateUserClusterRoleBinding(t Test, userName string, roleName string) *rba
 
 	return rb
 }
+
+func CreateUserNamespaceRoleBinding(t Test, userName string, namespace *corev1.Namespace, roleName string) *rbacv1.RoleBinding {
+	t.T().Helper()
+
+	// Create a RoleBinding to give specified role access to the user for test-namespace
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "namespace-admin-access",
+			Namespace: namespace.Name,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:     "User",
+				Name:     userName,
+				APIGroup: "rbac.authorization.k8s.io",
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "ClusterRole",
+			Name:     "admin", // grants admin access
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+
+	rb, err := t.Client().Core().RbacV1().RoleBindings(namespace.Name).Create(t.Ctx(), roleBinding, metav1.CreateOptions{})
+	t.Expect(err).NotTo(gomega.HaveOccurred())
+	t.T().Logf("Created User ClusterRoleBinding %s successfully", roleBinding.Name)
+
+	t.T().Cleanup(func() {
+		t.Client().Core().RbacV1().ClusterRoleBindings().Delete(t.Ctx(), rb.Name, metav1.DeleteOptions{})
+	})
```

Comment:
Dedicated cleanup isn't needed as the Role is created in test namespace which is cleaned up after test.